anywidget/widget.py

```diff
@@ -48,3 +51,25 @@ def __init__(self, *args, **kwargs) -> None:
         ).tag(sync=True)
 
         self.add_traits(**anywidget_traits)
+
+        # Check if we are in Colab
+        if "google.colab.output" in sys.modules:
+            # Enable custom widgets manager so that our widgets display in Colab
+            # https://github.com/googlecolab/colabtools/issues/498#issuecomment-998308485
+            if not type(self)._enabled_colab_widget_manager:
```

Comment:
it's not particularly better... but if you want to ensure that a function only gets called once, you can use `functools.lru_cache`

```python
@lru_cache(maxsize=None)
def _enable_custom_widget_manager():
    # Enable custom widgets manager so that our widgets display in Colab
    # https://github.com/googlecolab/colabtools/issues/498#issuecomment-998308485
    output = sys.modules.get("google.colab.output")
    output.enable_custom_widget_manager()  # type: ignore
```

and then 

```py
if "google.colab.output" in sys.modules:
    _enable_custom_widget_manager()
```

Reply:
ah ok... I actually did this with `functools.cache` originally, but then that failed for older versions for Python! I like this better.

Reply:
yeah, `cache` came later, but `lru_cache` is there for older versions